### PR TITLE
ITSASU-555: Self Employed CYA page

### PR DESCRIPTION
**Pull request by sb-dev**
Update Self Employed CYA page to confirm SelfEmploymentData when saving
it in the DB.

### CODE REVIEW

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/services/MultipleSelfEmploymentsService.scala [hunk 1]
```diff
@@ -38,7 +38,7 @@ class MultipleSelfEmploymentsService @Inject()(incomeTaxSubscriptionConnector: I
 
   def saveBusinessStartDate(businessId: String, businessStartDate: BusinessStartDate)
                            (implicit hc: HeaderCarrier): Future[Either[SaveSelfEmploymentDataFailure.type, PostSelfEmploymentsSuccess]] = {
-    saveData(businessId, _.copy(businessStartDate = Some(businessStartDate)))
+    saveData(businessId, _.copy(businessStartDate = Some(businessStartDate), confirmed = false))
```
Comment (scullythornton1) [inline app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/services/MultipleSelfEmploymentsService.scala:41]: hi Samir, can I know if those confirmed = false is necessary when it already defaults to false in the SelfEmploymentData model?

File: it/controllers/SelfEmployedCYAControllerISpec.scala [hunk 2]
```diff
@@ -134,6 +143,7 @@ class SelfEmployedCYAControllerISpec  extends ComponentSpecBase with FeatureSwit
       stubAuthSuccess()
       stubGetSelfEmployments(businessAccountingMethodKey)(OK, Json.toJson(testAccountingMethodModel))
       stubGetSelfEmployments(businessesKey)(OK, Json.toJson(testBusinesses))
+      stubSaveSelfEmployments(businessesKey, Json.toJson(testConfirmedBusinesses))(OK)
       And("save & retrieve feature switch is enabled")
       enable(SaveAndRetrieve)
```
Comment (scullythornton1) [inline it/controllers/SelfEmployedCYAControllerISpec.scala:149]: hi I know you did not write this part of the code, but can you help to amend 

in row 153, can you amend the description from "Should return a SEE_OTHER with a redirect location of confirmation" to "Should return a SEE_OTHER with a redirect location of task list page"

in row 171, do the same change the description to 
"Should return a SEE_OTHER with a redirect location of self-employed CYA page"

**Comment by scullythornton1** [comment]
can you also add a test when
 multipleSelfEmploymentsService.confirmBusiness(id)  map return left => throw new InternalServerException("[SelfEmployedCYAController][submit] - Failure to save self employment data")

thanks

File: app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/services/MultipleSelfEmploymentsService.scala (hunk 1 above)
Comment (sb-dev) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/services/MultipleSelfEmploymentsService.scala:41]: The `confirmed` flag gets set to true when the user confirms the data via the CYA page. It's necessary to reset the flag when users update any field.

Comment (scullythornton1) [reply app/uk/gov/hmrc/incometaxsubscriptionselfemployedfrontend/services/MultipleSelfEmploymentsService.scala:41]: I see, thanks

**Pull request merged**